pkg/cfn/template/api.go

```diff
@@ -31,11 +31,17 @@ type AnyResource struct {
 	Properties interface{}
 }
 
-// Output represents a CloudFormation output definition
-type Output struct {
-	Description string          `json:",omitempty"`
-	Value       MapOfInterfaces `json:",omitempty"`
-}
+type (
+	// Output represents a CloudFormation output definition
```

Comment:
What is an output definition? Like "json" would be one?

Reply:
Templates have the following main sections: `Parameters`, `Conditions`, `Mappings`, `Resources` and `Outputs`.

We only use `Resources` and `Outputs`.

https://github.com/weaveworks/eksctl/blob/f6a3000c2137485659eb0159736bace6784d593f/pkg/cfn/template/api.go#L19-L26

Reply:
Here is how we use outputs in cluster stack:
```JSON
{
  "ARN": {
    "Export": {
      "Name": {
        "Fn::Sub": "${AWS::StackName}::ARN"
      }
    },
    "Value": {
      "Fn::GetAtt": "ControlPlane.Arn"
    }
  },
  "CertificateAuthorityData": {
    "Value": {
      "Fn::GetAtt": "ControlPlane.CertificateAuthorityData"
    }
  },
  "ClusterStackName": {
    "Value": {
      "Ref": "AWS::StackName"
    }
  },
  "Endpoint": {
    "Export": {
      "Name": {
        "Fn::Sub": "${AWS::StackName}::Endpoint"
      }
    },
    "Value": {
      "Fn::GetAtt": "ControlPlane.Endpoint"
    }
  },
  "FeatureNATMode": {
    "Value": "Single"
  },
  "SecurityGroup": {
    "Export": {
      "Name": {
        "Fn::Sub": "${AWS::StackName}::SecurityGroup"
      }
    },
    "Value": {
      "Ref": "ControlPlaneSecurityGroup"
    }
  },
  "ServiceRoleARN": {
    "Export": {
      "Name": {
        "Fn::Sub": "${AWS::StackName}::ServiceRoleARN"
      }
    },
    "Value": {
      "Fn::GetAtt": "ServiceRole.Arn"
    }
  },
  "SharedNodeSecurityGroup": {
    "Export": {
      "Name": {
        "Fn::Sub": "${AWS::StackName}::SharedNodeSecurityGroup"
      }
    },
    "Value": {
      "Ref": "ClusterSharedNodeSecurityGroup"
    }
  },
  "SubnetsPrivate": {
    "Export": {
      "Name": {
        "Fn::Sub": "${AWS::StackName}::SubnetsPrivate"
      }
    },
    "Value": {
      "Fn::Join": [
        ",",
        [
          {
            "Ref": "SubnetPrivateUSWEST2B"
          },
          {
            "Ref": "SubnetPrivateUSWEST2C"
          },
          {
            "Ref": "SubnetPrivateUSWEST2D"
          }
        ]
      ]
    }
  },
  "SubnetsPublic": {
    "Export": {
      "Name": {
        "Fn::Sub": "${AWS::StackName}::SubnetsPublic"
      }
    },
    "Value": {
      "Fn::Join": [
        ",",
        [
          {
            "Ref": "SubnetPublicUSWEST2C"
          },
          {
            "Ref": "SubnetPublicUSWEST2D"
          },
          {
            "Ref": "SubnetPublicUSWEST2B"
          }
        ]
      ]
    }
  },
  "VPC": {
    "Export": {
      "Name": {
        "Fn::Sub": "${AWS::StackName}::VPC"
      }
    },
    "Value": {
      "Ref": "VPC"
    }
  }
}
```

Reply:
As in the `Outputs` section in [CloudFormation template](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/outputs-section-structure.html)